### PR TITLE
[DOXIA-740] Add test for non-consecutive headings leading to incomplete

### PR DESCRIPTION
**Pull request by kwin**
output

### CODE REVIEW

File: doxia-core/src/main/java/org/apache/maven/doxia/macro/toc/TocMacro.java [hunk 1]
```diff
@@ -112,9 +113,13 @@ public void execute(Sink sink, MacroRequest request) throws MacroExecutionExcept
             tocSink.close();
         }
 
-        IndexEntry index = tocSink.getRootEntry();
+        writeTocForIndexEntry(sink, getAttributesFromMap(request.getParameters()), tocSink.getRootEntry());
+    }
+
+    void writeTocForIndexEntry(Sink sink, SinkEventAttributes listAttributes, IndexEntry rootEntry) {
```
Comment (kwin) [inline doxia-core/src/main/java/org/apache/maven/doxia/macro/toc/TocMacro.java:119]: This still needs some work to
a) only consider entries for type section
b) skip empty entries (not having an id)

**Comment by kwin** [comment]
@michael-o Feel free to already have a look, although this need to consider some more edge cases (see comments inline).

File: doxia-core/src/main/java/org/apache/maven/doxia/macro/toc/TocMacro.java (hunk 1 above)
Comment (kwin) [reply doxia-core/src/main/java/org/apache/maven/doxia/macro/toc/TocMacro.java:119]: Done now.

File: doxia-core/src/main/java/org/apache/maven/doxia/index/IndexingSink.java [hunk 2]
```diff
@@ -150,14 +158,20 @@ public void text(String text, SinkEventAttributes attributes) {
...
+                case SECTION_6:
                     // -----------------------------------------------------------------------
                     // Sanitize the id. The most important step is to remove any blanks
                     // -----------------------------------------------------------------------
 
                     // append text to current entry
                     IndexEntry entry = stack.lastElement();
 
-                    String title = entry.getTitle() + text;
+                    String title;
```
Comment (michael-o) [inline doxia-core/src/main/java/org/apache/maven/doxia/index/IndexingSink.java:169]: Why not `title = entry.getTitle()` and then `title += text`/`title = text`?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: doxia-core/src/main/java/org/apache/maven/doxia/index/IndexingSink.java (hunk 2 above)
Comment (kwin) [reply doxia-core/src/main/java/org/apache/maven/doxia/index/IndexingSink.java:169]: Because entry.getTitle() might return null

Comment (michael-o) [reply doxia-core/src/main/java/org/apache/maven/doxia/index/IndexingSink.java:169]: I mean for the if else clause:
```
String title = entry.getTitle();
                    if (title != null) {
                        title += text;
                    } else {
                        title = text;
                    }
```

Comment (kwin) [reply doxia-core/src/main/java/org/apache/maven/doxia/index/IndexingSink.java]: Done in https://github.com/apache/maven-doxia/pull/228/commits/bd1e61ea7705af4a6a6f19ad28355a5c6807c343.

**Review by michael-o — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #866

**Pull request merged**